hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -344,8 +344,10 @@ public Partitioner getUpsertPartitioner(WorkloadProfile profile) {
 
         // TODO - check if index.isglobal then small files are log files too
         Optional<FileSlice> smallFileSlice = getRTFileSystemView()
-            .getLatestFileSlicesBeforeOrOn(partitionPath, latestCommitTime.getTimestamp()).filter(
+            // Use the merged file-slice for small file selection
+            .getLatestMergedFileSlicesBeforeOrOn(partitionPath, latestCommitTime.getTimestamp()).filter(
                 fileSlice -> fileSlice.getLogFiles().count() < 1
+                    // TODO: Fix the assumption about data-file. Will be fixed by Nishith's insert to log-files PR
```

Comment:
May be remove this comment if the issue is addressed.

Reply:
@n3nash : Done. I checked that this is addressed in https://github.com/uber/hudi/pull/413